resources/views/projects/tasks/tab.blade.php

```diff
@@ -0,0 +1,285 @@
+			const taskId = $(e.target).closest(".task").data("taskid");
+			showTaskDetailsModal(taskId);
+		});
+
+		/* format datetimes */
+		renderDateTimeAgoOnce();
+		utcToLocal();
+	@endif
+	@if($project->isCollaborator(Auth::id()))
+		function showTaskModal(action) {
+			$(`#${action}-task-form-modal`).modal("show");
+		}
+
+		function submitTaskForm(action) {
+			const $form = $(`#${action}-task-form`);
+			let taskFields;
+			let method;
+			let url;
+			let data;
+
+			if(action == "new") {
+				taskFields = {
+					title: ["empty", "maxLength[255]"],
+					description: ["empty"],
+					dueDate: ["empty"]
+				};
+			}
+			else { //edit
+				taskFields = {
+					title: ["empty", "maxLength[255]"],
+					description: ["empty"],
+					dueDate: ["empty"],
+					status: ["minCount[1]"]
+				};
+			}
+
+			$(`#${action}-task-form`).form({
+				fields: taskFields,
+				onSuccess: function(event) {
+					event.preventDefault();
+
+					// shared values
+					let title = $form.find("input[name=title]").val();
+					let dueDate = $form.find("input[name=dueDate]").val();
+					let isoDueDate = new Date(dueDate).toISOString();
+					let description = $form.find("textarea[name=description]").val();
+
+					if(action == "new") {
+						method = "POST";
+						url = "/tasks";
+						data = {
+							'title': title,
+							'dueDate': isoDueDate,
+							'description': description,
+							'project': $form.find("input[name=project]").val()
+						};
+					}
+					else { //edit
+						method = "PATCH";
+						let taskId = $form.find("input[name=task-id]").val();
+						url = `/tasks/${taskId}`;
+						data = {
+							'title': title,
+							'dueDate': isoDueDate,
+							'description': description,
+							'task_status' : $form.find("select[name=status]").val()
+						};
+					}
+
+					$.ajax({
+						type: method,
+						url: url,
+						headers: {
+							'X-CSRF-TOKEN': $('meta[name="csrf-token"]').attr('content')
+						},
+						data: data,
+						dataType: 'json',
+						success: function (task) {
+							taskToAdd = getTaskComponent(task);
+
+							if(action == "edit") {
+								$(".task").filter(`[data-taskid="${task.id}"]`).remove();
+							}
+
+							let taskList = "";
+
+							switch(task.task_status) {
+								case 1: //todo
+									taskList = "to-do";
+									break;
+								case 2: // in-progress
+									taskList = "in-progress";
+									break;
+								case 3: // closed
+									taskList = "closed";
+									break;
+							}
+
+							$(`#${taskList}-tasks > .task`).first().before(taskToAdd);
+
+							// format datetimes
+							renderDateTimeAgoOnce();
+							utcToLocal();
+
+							$(".task-form-modal").modal("hide");
+						},
+						error: function () {
+							$(".task-form-modal").modal("hide");
+							$(`#${action}-task-form-error-modal`).modal("show");
+						}
+					});
+				},
+				onFailure: function() {
+					return false;
+				}
+			});
+
+			$(`#${action}-task-form`).submit();
```

Comment:
Si est&aacute;s haciendo esto via AJAX arriba, para qu&eacute; lo submiteas otra vez?

Reply:
Sólo se submitea una vez. Necesito el ` event.preventDefault(); ` que está arriba porque de lo contrario semantic va a hacer force reload cuando se submitee. Por eso yo la tengo que submittear manualmente. 

En milestones hicieron lo mismo (although it beats the purpose there porque igual hacemos reload ahí).

Reply:
Still, cual es el punto de submitearlo?